pkg/api/types.go

```diff
@@ -1901,6 +1901,7 @@ type PodAffinityTerm struct {
 	// A label query over a set of resources, in this case pods.
 	// +optional
 	LabelSelector *metav1.LabelSelector
+	// Deprecated: please use the NamespaceSelector, this field will be removed in 1.8.
```

Comment:
Is that what we really want?  This issue links to 2 issues back but doesn't contain the concise reasoning on why you want to special case the selector.